# 机械臂构型参数

# 7轴机械臂关节范围约束
dh_parameters:
  d_bs: 0.1583          # 基座到肩部偏移 (m)
  d_se: 0.3596          # 肩部到肘部长度 (m)
  d_ew: 0.2569          # 肘部到腕部长度 (m)
  a_wf: 0.012           # 腕部到工具端x偏移 (m)
  a_se: 0.009           # 肩部x方向偏移 (m)
  a_ee: 0.0395          # 工具端偏移 (m)

# 驱动转化参数 (Conversion Drive Parameters)
Conversion_drive_params:
  l_bx: 0.2146
  l_by: 0.015
  l_ofsx: 0.0234
  l_ofsy: 0.015
  beta40_degrees: 28.66

  # 推杆初始长度 (Rod Init Lengths)
motor_init_lengths:
  l_m10: 0.1912
  l_m20: 0.15153
  l_m30: 0.15153

# 腕关节计算参数 (Wrist Joint Calculation Parameters)
wrist_joint_params:
  d_cx: 0.012
  d_cy: 0.028
  d_cz: 0.019
  d_ax: 0.0
  d_ay: 0.031
  d_az: 0.1503


# 关节限制 (Joint Limits)
# 推荐使用列表的列表或列表的映射，更清晰
joint_limits:
  - {min: -1.57079632679, max: 1.57079632679} # -PI/2 到 PI/2
  - {min: -3.14159265359, max: 0.1} # -PI 到 0.1
  - {min: -3.14159265359, max: 3.14159265359} # -PI 到 PI
  - {min: 0.0, max: 2.26}
  - {min: -3.14159265359, max: 3.14159265359} # -PI 到 PI
  - {min: -1.22, max: 1.22}
  - {min: -0.5235987756, max: 0.5235987756} # -PI/6 到 PI/6